webcompat/templates/__init__.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Module where we register Jinja template filters."""
+
+import hashlib
+import os
+import re
+
+from flask import Markup
+from webcompat import app
+
+STATIC_PATH = os.getcwd() + '/webcompat/static'
+
+cache_dict = {}
+
+
+@app.template_filter('bust_cache')
+def bust_cache(file_path):
+    """Jinja2 filter to add a cache busting param based on md5 checksum.
+
+    Uses a simple cache_dict to we don't have to hash each file for every
+    request. This is kept in-memory so it will be blown away when the app
+    is restarted (which is when file changes would have been deployed).
+    """
+    def get_checksum(file_path):
+        try:
+            checksum = cache_dict[file_path]
+        except KeyError:
+            checksum = md5_checksum(file_path)
+            cache_dict[file_path] = checksum
+        return checksum
+
+    return file_path + '?' + get_checksum(STATIC_PATH + file_path)
+
+
+def md5_checksum(file_path):
+    """Return the md5 checksum for a given file path."""
+    with open(file_path, 'rb') as fh:
+        m = hashlib.md5()
+        while True:
+            # only read in 8k of the file at a time
+            data = fh.read(8192)
+            if not data:
+                break
+            m.update(data)
+        return m.hexdigest()
+
+
+@app.template_filter('format_date')
+def format_date(datestring):
+    """For now, just chops off crap."""
+    # 2014-05-01T02:26:28Z
+    return datestring[0:10]
+
+
+@app.template_filter('format_title')
```

Comment:
Here's the source I was porting from https://github.com/webcompat/webcompat.com/blob/c34522f95a14ed6efa903a40e1fc8902c2d2b11e/webcompat/static/js/lib/models/issue.js#L73

It's mostly the same, except for `get_description` -- since it was running in the context of JS / DOM, it was simpler to grab text w/o tags (via `textConent` to run the regexp on. Luckily, flask provides a `striptags` method: https://flask.palletsprojects.com/en/1.1.x/api/#flask.Markup.striptags

Reply:
I'm closely following what you are doing these days and loving it.